### PR TITLE
fabric-chaincode-haskell

### PR DESCRIPTION
**Pull request by airvin**
`fabric-chaincode-haskell` is a Haskell shim for Hyperledger Fabric to allow the authoring of smart contracts in Haskell.

As chaincode often contains business and security critical code, it is important that the code is correct. Implications for contracts containing bugs and exploits can lead to attacks like the DAO and Parity attacks on the Ethereum blockchain, which can have significant economic impacts. Functionally pure and strongly typed languages like Haskell provide a much greater degree of confidence in correctness. `fabric-chaincode-haskell` allows the authorship of chaincode that gives a higher confidence of correctness.

Allowing chaincode to be authored in Haskell also opens up the world of blockchain to the large and enthusiastic community of Haskell developers.

### CODE REVIEW

**Comment by lehors** [comment]
LGTM. I support the creation of this lab.

File: labs/fabric-chaincode-haskell.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+- [nwaywood](https://github.com/nwaywood) - Nick Waywood
+- [airvin](https://github.com/airvin) - Allison Irvin
+
+# Sponsor
+
+- Arnaud Le Hors (lehors@us.ibm.com) - Chair of Hyperledger TSC 
+
+# Pre-existing repository
+
+- https://github.com/nwaywood/fabric-chaincode-haskell
```
Comment (tkuhrt) [inline labs/fabric-chaincode-haskell.md:31]: It does not appear that you have DCO signoff in this repository. We will not be able to transfer as is.

**Comment by vipinsun** [comment]
So are we waiting for DCO on incoming lab or creating this? If DCO on incoming lab is necessary, then we should augment DCO check to add the other lab as well. We have some suggestions for squashing and committing in case the original contributors are not available

**Comment by lehors** [comment]
The DCO issue on the existing lab should be easy to resolve because @airvin and @nwaywood are the only ones who have made commits.

**Comment by airvin** [comment]
Thanks @lehors, @tkuhrt and @vipinsun.

We have done an interactive rebase of the repository and added the `Signed-off-by` to each of the commits. Is it okay to transfer this repository to Hyperledger Labs the way it is now?

**Comment by tkuhrt** [comment]
@ryjones : Can you help us with this repo transfer too?

**Comment by ryjones** [comment]
@tkuhrt yes - once this is merged, @nwaywood, please transfer the repo to me (@ryjones) and I will put it in place.

**Comment by nwaywood** [comment]
@ryjones I just submitted the transfer request 👍

**Pull request merged**